src/services/NumberService.ts

```diff
@@ -55,15 +60,25 @@ export class TestNumberService {
     }
 
     /**
-     * Retrieves the last test number
+     * Retrieves the last test number or trailerId based on the numberType param
+     * @param numberType - type of number to be retrieved from the DB - default is testNumber
      */
-    public getLastTestNumber(): Promise<TestNumber> {
+    public getLastNumber(numberType: NUMBER_TYPE.TRAILER_ID | NUMBER_TYPE.TEST_NUMBER = NUMBER_TYPE.TEST_NUMBER): Promise<TestNumber | TrailerId> {
         return this.dbClient.scan()
             .then((data: any) => {
                 if (data.Count === 0) {
+                    if (numberType === NUMBER_TYPE.TRAILER_ID) {
```

Comment:
I don't prefer the approach of conditional logic. We may have to do another ticket where we have to use this service for system number generation. I would suggest to have separate `getLastTrailerId()` as a wrapping function and common logic used by both `getLastTestNumber` and `getLastTrailerId()`.